pkg/bindings/images/build.go

```diff
@@ -557,261 +581,243 @@ func Build(ctx context.Context, containerFiles []string, options types.BuildOpti
+	}
+	// Multipart request structure:
+	// - "MainContext": The main build context as a tar file
+	// - "build-context-<name>": Each additional local context as a tar file
+	logrus.Debugf("Using additional local build contexts: %v", localContexts)
+	pr, pw := io.Pipe()
+	writer := multipart.NewWriter(pw)
+	contentType = writer.FormDataContentType()
+	requestParts.Body = pr
 
-				go func() {
-					defer pw.Close()
-					defer writer.Close()
+	if requestParts.Headers == nil {
+		requestParts.Headers = make(http.Header)
+	}
+	requestParts.Headers.Set("Content-Type", contentType)
 
-					mainContext, err := writer.CreateFormFile("MainContext", "MainContext.tar")
-					if err != nil {
-						pw.CloseWithError(fmt.Errorf("creating form file for main context: %w", err))
-						return
-					}
+	go func() {
+		defer pw.Close()
+		defer writer.Close()
 
-					if _, err := io.Copy(mainContext, tarfile); err != nil {
-						pw.CloseWithError(fmt.Errorf("copying main context: %w", err))
-						return
-					}
+		mainContext, err := writer.CreateFormFile("MainContext", "MainContext.tar")
+		if err != nil {
+			pw.CloseWithError(fmt.Errorf("creating form file for main context: %w", err))
+			return
+		}
 
-					for name, context := range localContexts {
-						logrus.Debugf("Processing additional local context: %s", name)
-						part, err := writer.CreateFormFile(fmt.Sprintf("build-context-%s", name), name)
-						if err != nil {
-							pw.CloseWithError(fmt.Errorf("creating form file for context %q: %w", name, err))
-							return
-						}
+		if _, err := io.Copy(mainContext, tarfile); err != nil {
+			pw.CloseWithError(fmt.Errorf("copying main context: %w", err))
+			return
+		}
 
-						// Context is already a tar
-						if archive.IsArchivePath(context.Value) {
-							file, err := os.Open(context.Value)
-							if err != nil {
-								pw.CloseWithError(fmt.Errorf("opening archive %q: %w", name, err))
-								return
-							}
-							if _, err := io.Copy(part, file); err != nil {
-								file.Close()
-								pw.CloseWithError(fmt.Errorf("copying context %q: %w", name, err))
-								return
-							}
-							file.Close()
-						} else {
-							tarContent, err := nTar(nil, context.Value)
-							if err != nil {
-								pw.CloseWithError(fmt.Errorf("creating tar content %q: %w", name, err))
-								return
-							}
-							if _, err = io.Copy(part, tarContent); err != nil {
-								pw.CloseWithError(fmt.Errorf("copying tar content %q: %w", name, err))
-								return
-							}
-							if err := tarContent.Close(); err != nil {
-								logrus.Errorf("Error closing tar content for context %q: %v\n", name, err)
-							}
-						}
-					}
-				}()
-				logrus.Debugf("Multipart body is created with content type: %s", contentType)
-			} else {
-				requestBody = tarfile
-				logrus.Debugf("Using main build context: %q", options.ContextDirectory)
+		defer func() {
+			if err := tarfile.Close(); err != nil {
+				logrus.Errorf("%v\n", err)
```

Comment:
Maybe wrap this error with helper text.

Reply:
I included helper text for this error message.